packages/gestalt/src/ModalAlert/Action.js

```diff
@@ -1,66 +1,56 @@
 // @flow strict
 import { type Node } from 'react';
 import Button from '../Button.js';
+import ButtonLink from '../ButtonLink.js';
 
-export type ActionDataType = {|
+type LinkDataType = {|
+  role: 'link',
   accessibilityLabel: string,
   disabled?: boolean,
   href?: string,
   label: string,
-  onClick?: ({|
-    event:
-      | SyntheticMouseEvent<HTMLButtonElement>
-      | SyntheticMouseEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLButtonElement>,
-    dangerouslyDisableOnNavigation: () => void,
-  |}) => void,
+  onClick?: $ElementType<React$ElementConfig<typeof ButtonLink>, 'onClick'>,
   rel?: 'none' | 'nofollow',
   target?: null | 'self' | 'blank',
 |};
+type ButtonDataType = {|
+  role: 'button',
+  accessibilityLabel: string,
+  disabled?: boolean,
+  label: string,
+  onClick?: $ElementType<React$ElementConfig<typeof Button>, 'onClick'>,
+|};
 
-type Props = {| ...ActionDataType, dataTestId?: string, type: string |};
+type Props = {| ...LinkDataType | ButtonDataType, dataTestId?: string, type: string |};
```

Comment:
I don't like this type solution.



why don't we do ?

`
{| role = 'link', ...|} | {| role = 'button', ... |} 
`

Reply:
or simply `type Props = LinkDataType | ButtonDataType

Reply:
DONE (as `type Props = LinkDataType | ButtonDataType`)